### PR TITLE
Install dependencies for Open ID Connect implementation for MISP v2.4.140

### PR DESCRIPTION
**Pull request by garrit-schroeder**
Hello,

with MISP version 2.4.140 open id connect was implemented. This pull request installs the necessary dependencies and modifies the nginx files to allow integration.

Are you interested in adding this to your repo?
If yes I am more than happy to write a little setup guide in the readme file.

Tested with Keycloak as IDP.

### CODE REVIEW

**Comment by garrit-schroeder** [comment]
Modification in config.php

```
'Security' => 
  array (
    ...
    array (
      0 => 'OidcAuth.Oidc',
    ),
  ),
  'OidcAuth' => [
        'provider_url' => 'https://KEYCLOAK/auth/realms/REALM/.well-known/openid-configuration',
        'client_id' => 'mymisp',
        'client_secret' => 'XXX',
        'role_mapper' => [ // if user has multiple roles, first role that match will be assigned to user
            'user' => 3, // User
            'admin' => 1, // Admin
        ],
        'default_org' => 'default',
    ],

**Comment by coolacid** [comment]
Awesome! Looks like a simple enough change. Give me a day or two to review. Glad I didn't tag the 140 build yet.

File: server/Dockerfile [hunk 1]
```diff
@@ -153,6 +154,9 @@ ARG PHP_VER
 # Make a copy of the configurations, so we can sync from it
     RUN cp -R /var/www/MISP/app/Config /var/www/MISP/app/Config.dist
 
+    # Intall Open ID Connect dependency
+    RUN cd /var/www/MISP/app && php composer.phar require jumbojett/openid-connect-php
```
Comment (garrit-schroeder) [inline server/Dockerfile:158]: This should probably be done differently. But how?

Comment (coolacid) [reply server/Dockerfile:158]: Maybe append the file here: https://github.com/coolacid/docker-misp/blob/9977d9281ffe5e843cd3a0ee73601c9158c2dbe3/server/Dockerfile#L4

Comment (garrit-schroeder) [reply server/Dockerfile]: Done

Comment (coolacid) [reply server/Dockerfile]: I wonder if a PR should be made to MISP to include this requirement in the composer.json file?

**Comment by garrit-schroeder** [comment]
> Modification in config.php
> 
> ```
> 'Security' => 
>   array (
>     ...
>     'auth' => array (
>       0 => 'OidcAuth.Oidc',
>     ),
>   ),
> 'OidcAuth' => [
>     'provider_url' => 'https://KEYCLOAK/auth/realms/REALM/.well-known/openid-configuration',
>     'client_id' => 'mymisp',
>     'client_secret' => 'XXX',
>     'role_mapper' => [ // if user has multiple roles, first role that match will be assigned to user
>         'user' => 3, // User
>         'admin' => 1, // Admin
>     ],
>     'default_org' => 'default',
> ],
> ```

Changing a MISP setting in Server Settings & Maintenance will remove this block from the config.php file. Do you have any idea why MISP is doing that?

File: server/Dockerfile (hunk 1 above)
Comment (garrit-schroeder) [reply server/Dockerfile]: Its listed as suggested dependency already.

https://github.com/MISP/MISP/blob/2.4/app/composer.json

Comment (coolacid) [reply server/Dockerfile]: Ah, Ok I haven't had to work with composer in many many years. This looks like the right way to do this.

**Comment by coolacid** [comment]
Going to wait for the MISP bug to be looked at.

**Comment by garrit-schroeder** [comment]
Tested today again with MISP version 141 and it is working now. The issue is resolved.

**Comment by coolacid** [comment]
Merged. I'll look to cut a `b` release later unless MISP updates sooner.

**Pull request merged**